.github/workflows/ci-pr-title.yaml

```
@@ -0,0 +1,90 @@
+name: CI Check Title
+
+on:
+  pull_request:
+    types: [opened, edited, synchronize, reopened]
+
+jobs:
+  title-lint:
+    name: Validate PR title
+    runs-on: [default]
+    steps:
+      - name: CI Check Title
+        uses: amannn/action-semantic-pull-request@v5
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          wip: true
+          # Configure which types are allowed (newline-delimited).
+          # Default: https://github.com/commitizen/conventional-commit-types
+          types: |
+            build
+            chore
+            fix
+            feat
+            merge
+            publish
+            release
+            refactor
+            research
+            style
+            test
+          # Configure which scopes are allowed (newline-delimited).
+          # These are regex patterns auto-wrapped in `^ $`.
+          scopes: |
+            cluster-controller
+            organization-controller
+            plugin-controller
+            plugindefinition-controller
+            team-controller
+            teammembership-controller
+            teamsrbac-controller
+            build
+            config
+            ci
+            core
```

A generic controller for changes to lifecycle or other common parts
```suggestion
            core
            controller
```